### PR TITLE
fix: Support AWS regions

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?

Changes multiple elements to support AWS regions:
* Read replicas now point to the correct server and set the correct module name
* Replica repository is loaded dynamically and only when required to avoid extra modules to be loaded

### CODE REVIEW

File: lib/realtime/api.ex [hunk 1]
```diff
@@ -6,7 +6,13 @@ defmodule Realtime.Api do
...
   import Ecto.Query
 
-  alias Realtime.{Repo, Api.Tenant, Api.Extensions, RateCounter, GenCounter, Tenants}
+  alias Realtime.Repo
+  alias Realtime.Repo.Replica
+  alias Realtime.Api.Tenant
+  alias Realtime.Api.Extensions
+  alias Realtime.RateCounter
+  alias Realtime.GenCounter
+  alias Realtime.Tenants
```
Comment (abc3) [inline lib/realtime/api.ex:15]: I don't think this is necessary, realtime repo mainly uses a multi-alias style

File: lib/realtime/postgres_cdc.ex [hunk 2]
```diff
@@ -60,23 +60,35 @@ defmodule Realtime.PostgresCdc do
...
+        "ap-southeast-2" -> "syd"
+        "ap-south-1" -> "syd"
+        "eu-west-1" -> "lhr"
+        "eu-west-2" -> "lhr"
+        "eu-west-3" -> "lhr"
+        "eu-central-1" -> "lhr"
+        _ -> nil
+      end
+    else
+      aws_region
```
Comment (abc3) [inline lib/realtime/postgres_cdc.ex:91]: Do we want to have RT nodes in all Supabase regions?

File: lib/realtime/repo_replica.ex [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      is_nil(replica_conf) ->
+        Logger.error("Replica config not found for #{region} region")
+        Realtime.Repo
+
+      true ->
+        # Check if module is present
+        case Code.ensure_compiled(replica) do
+          {:module, _} -> nil
+          _ -> {:module, _, _, _} = Module.create(replica, @ast, Macro.Env.location(__ENV__))
+        end
```
Comment (abc3) [inline lib/realtime/repo_replica.ex:64]: Is there a way to avoid dynamic creation of modules?

File: lib/realtime/api.ex (hunk 1 above)
Comment (filipecabaco) [reply lib/realtime/api.ex:15]: This was the thing I mentioned during the sync that makes it trickier to find module usage within the code (e.g. with Realtime.Api.Tenant it becomes way easier to find aliases vs Api.Tenant in one module and Tenant in another module)

File: lib/realtime/postgres_cdc.ex (hunk 2 above)
Comment (filipecabaco) [reply lib/realtime/postgres_cdc.ex:91]: doubt it and we could a check just to validate that we're in an allowed region

something along the lines of 
```elixir
aws_region in ["us-east-1", "eu-west-1", ...]
```

File: lib/realtime/repo_replica.ex (hunk 3 above)
Comment (filipecabaco) [reply lib/realtime/repo_replica.ex:64]: trickier, we would need to define the region at compile time so we could properly load things up. with this change we're using runtime stuff which makes it trickier.

plus, with the current implementation, we are creating more repo modules as we are creating a Repo per region even if they are not used

File: lib/realtime/postgres_cdc.ex (hunk 2 above)
Comment (abc3) [reply lib/realtime/postgres_cdc.ex:91]: `aws_region` is the region where the tenant's db resides. This function is responsible for initiating a database connection in the closest region. For example, if a tenant's database is located in aws's `us-east-1`, then we should establish a connection on a node in fly's `iad` region.
```elixir
"us-east-1" -> "iad"
```

File: lib/realtime/api.ex (hunk 1 above)
Comment (abc3) [reply lib/realtime/api.ex:15]: many apps use a multi alias style, such as [phoenix](https://github.com/phoenixframework/phoenix/blob/1c1d50e36a9a293c9649c1649cae57dfa386e328/lib/phoenix/socket.ex#L201)

Comment (abc3) [reply lib/realtime/api.ex:15]: But, yeah, if it makes it harder to read, feel free to change that. However, I don't see any difference, just more lines

File: lib/realtime/postgres_cdc.ex (hunk 2 above)
Comment (filipecabaco) [reply lib/realtime/postgres_cdc.ex:91]: Yeah and you're right for some reason I had the assumption that we were deploying in every Supabase tenant region and use said RDS conn.

Will check what can be done

File: lib/realtime/repo_replica.ex [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    "lhr" => Realtime.Repo.Replica.FRA,
+    "fra" => Realtime.Repo.Replica.FRA
+  }
+
+  @replicas_aws %{
+    "ap-southeast-1" => Realtime.Repo.Replica.Singapore,
+    "ap-southeast-2" => Realtime.Repo.Replica.Sydney,
+    "eu-west-1" => Realtime.Repo.Replica.Ireland,
+    "eu-west-2" => Realtime.Repo.Replica.London,
+    "us-east-1" => Realtime.Repo.Replica.NorthVirginia
```
[the remark this answers is not in the document]
Comment (filipecabaco) [reply lib/realtime/repo_replica.ex]: yes it was 😅

**Review by abc3 — APPROVED** [review]
LGTM

**Pull request merged**